include/spdlog/sinks/qt_sinks.h

```diff
@@ -4,122 +4,81 @@
 #pragma once
 
 //
-// Custom sink for QPlainTextEdit or QTextEdit and its childs(QTextBrowser... etc)
-// Building and using requires Qt library.
+// Custom sink for QPlainTextEdit or QTextEdit and its childs(QTextBrowser...
+// etc) Building and using requires Qt library.
 //
 
 #include "spdlog/common.h"
 #include "spdlog/details/log_msg.h"
 #include "spdlog/details/synchronous_factory.h"
 #include "spdlog/sinks/base_sink.h"
 
+#include <type_traits>
 #include <QTextEdit>
 #include <QPlainTextEdit>
 
 //
-// qtextedit_sink class
+// qt_sink class
 //
 namespace spdlog {
 namespace sinks {
-template <typename Mutex>
-class qtextedit_sink : public base_sink<Mutex> {
+template <typename Mutex> class qt_sink : public base_sink<Mutex> {
 public:
-  qtextedit_sink(QTextEdit *textedit = nullptr) {
-    if (textedit != nullptr) {
-        qtextedit_ = textedit;
+  qt_sink(QObject *qt_object = nullptr, const std::string &meta_method = "") {
+    if (qt_object != nullptr) {
```

Comment:
i dont understand why to allow default nullptr in ctor if it get thrown afgerward if nullptr

Reply:
Actually it doesn't matter(nullptr or qobject*) in last solution 🤔

Reply:
why not?

Reply:
I think pass nullptr into QMetaObject::invokemethod function is valid but does nothing with nullptr. But with a valid QObject*  object it will invoke a function named in meta_method string.